File: src/chain/examples/idvalidation/SelfieMatchValidator.java
package chain.examples.idvalidation;

public class SelfieMatchValidator extends IdValidator {
    @Override protected boolean check(IdDocument doc) {
        return doc.getSelfieHash() != null
                && doc.getScanHash() != null
                && doc.getSelfieHash().charAt(0) == doc.getScanHash().charAt(0);
    }
    @Override protected String name() { return "SelfieMatchValidator"; }
}

File: src/chain/examples/idvalidation/SelfieMatchValidator.js
'use strict';

const IdValidator = require('./IdValidator');

class SelfieMatchValidator extends IdValidator {
  check(doc) {
    return !!doc.selfieHash && !!doc.scanHash && doc.selfieHash[0] === doc.scanHash[0];
  }
  name() { return 'SelfieMatchValidator'; }
}

module.exports = SelfieMatchValidator;
